test/test_pattern_matcher.py

```diff
@@ -17,6 +17,56 @@ def test_simple_match(self):
     self.assertEqual(matcher.rewrite(c1), c1)
     self.assertEqual(matcher.rewrite(c2), None)
 
+  def test_uop(self):
+    matcher = PatternMatcher([({"__name__": "x", "uop": UOps.CONST}, lambda x: x)])
+    c1 = UOp(UOps.CONST, dtypes.float, arg=1.0)
+    c2 = UOp(UOps.ALU, dtypes.float, (c1, c1), BinaryOps.ADD)
+    self.assertEqual(matcher.rewrite(c1), c1)
+    self.assertEqual(matcher.rewrite(c2), None)
+
+  def test_uop_set(self):
+    matcher = PatternMatcher([({"__name__": "x", "uop": {UOps.CONST, UOps.DEFINE_ACC}}, lambda x: x)])
+    c1 = UOp(UOps.CONST, dtypes.float, arg=1.0)
+    c2 = UOp(UOps.ALU, dtypes.float, (c1, c1), BinaryOps.ADD)
+    c3 = UOp(UOps.DEFINE_ACC, dtypes.int, arg=(1,))
```

Comment:
did you check how many args DEFINE_ACC has?
https://github.com/tinygrad/tinygrad/blob/master/tinygrad/codegen/linearizer.py#L106t

Reply:
i initially just ran all my defined UOps through the `type_verify` function and assumed that would be enough, my mistake. just switched to use mainly CONST and ALU ops to simplify things